tests/functional/test_htsget.py

```diff
@@ -0,0 +1,141 @@
+
+    output_dir = _run_pyega3(file_id, file_format, ref, start, end)
+
+    _assert_correct_file_is_downloaded(f'{output_dir}/{file_id}',
+                                       f'EE-2564.NA18636.alt_bwamem_GRCh38DH.20150826.CHB.exome_genomic_range_{ref}_{start}_{end}.cram',
+                                       '86b66eb1a56aaa0535a5aaecf8923552')
+
+    cleanup(output_dir)
+
+
+def test_htsget_bam():
+    file_id, file_format, ref = ['EGAF00007243773', 'BAM', 'chr10']
+
+    output_dir = _run_pyega3(file_id, file_format, ref)
+
+    _assert_correct_file_is_downloaded(f'{output_dir}/{file_id}',
+                                       f'HG00096.GRCh38DH__1097r__10.10000-10100__21.5000000-5050000_genomic_range_{ref}_None_None.bam',
+                                       'a5483f1da3981851ac0a5e4dddd2062f')
+
+    cleanup(output_dir)
+
+
+def test_htsget_bam_with_range():
+    file_accession, file_format, ref, start, end = ['EGAF00007243773', 'BAM', 'chr10', 10000, 10050]
+
+    output_dir = _run_pyega3(file_accession, file_format, ref, start, end)
+
+    _assert_correct_file_is_downloaded(f'{output_dir}/{file_accession}',
+                                       f'HG00096.GRCh38DH__1097r__10.10000-10100__21.5000000-5050000_genomic_range_{ref}_{start}_{end}.bam',
+                                       'a5483f1da3981851ac0a5e4dddd2062f')
+
+    cleanup(output_dir)
+
+
+def test_htsget_bcf():
+    file_accession, file_format, ref = ['EGAF00005001625', 'BCF', '22']
+
+    output_dir = _run_pyega3(file_accession, file_format, ref)
+
+    _assert_correct_file_is_downloaded(f'{output_dir}/{file_accession}',
+                                       f'ALL_chr22_20130502_2504Individuals_genomic_range_{ref}_None_None.bcf',
+                                       'c65ca1a4abd55351598ccbc65ebfa9a6')
+
+    cleanup(output_dir)
+
+
+def test_htsget_bcf_with_range():
+    file_accession, file_format, ref, start, end = ['EGAF00005001625', 'BCF', '22', 0, 100000]
+
+    output_dir = _run_pyega3(file_accession, file_format, ref, start, end)
+
+    _assert_correct_file_is_downloaded(f'{output_dir}/{file_accession}',
+                                       f'ALL_chr22_20130502_2504Individuals_genomic_range_{ref}_{start}_{end}.bcf',
+                                       'c65ca1a4abd55351598ccbc65ebfa9a6')
+
+    cleanup(output_dir)
+
+
+def _run_pyega3(file_accession: str, file_format: str, ref: str, start=None, end=None):
+    start_arg = '' if start is None else f'--start {start}'
+    end_arg = '' if end is None else f'--end {end} '
+    range_args = f'{start_arg} {end_arg}' if start_arg and end_arg else ''
+    outdir = _setup_output_dir()
+    cmd = f'pyega3 -t -d fetch {file_accession} --reference-name {ref} {range_args}--format {file_format} --output-dir {outdir}'
+    run_command_and_assert_download_complete(cmd)
+    return outdir
+
+
+def _setup_output_dir():
+    dir_id = str(uuid.uuid1())
+    output_dir = f'{script_dir}/.tmp/{dir_id}'
```

Comment:
I'd use a directory which is outside of the source code. See https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

Reply:
That's a good suggestion, I've applied this, thanks!